### ISSUE TITLE
Revistit constrastive finetuner

### ISSUE THREAD

**Issue by koaning** [koaning/embetter#73]
It seems I [glanced over something](https://sbert.net/docs/training/overview.html#loss-functions), which might help explain the benchmarks. 

> For each sentence pair, we pass sentence A and sentence B through our network which yields the embeddings u und v. The similarity of these embeddings is computed using cosine similarity and the result is compared to the gold similarity score. This allows our network to be fine-tuned and to recognize the similarity of sentences.

It's using _cosine similarity_ when it's comparing against the similarity ... which isn't what we are doing.

**Comment by koaning** [comment]
It also totally makes sense. If you're interested in "search" then you want to make sure that you're able to leverage cosine similarity. That also means that you gotta use it during training. 

This line [needs updating](https://github.com/koaning/embetter/blob/main/embetter/finetune/_contrastive.py#L142C52-L142C52).
